.github/workflows/ci.yml

```diff
@@ -178,6 +178,94 @@ jobs:
           if [ -f /Library/Filesystems/osxfusefs.fs/Support/load_osxfusefs ]; then /Library/Filesystems/osxfusefs.fs/Support/load_osxfusefs; elif [ -f /Library/Filesystems/osxfuse.fs/Contents/Resources/load_osxfuse ]; then /Library/Filesystems/osxfuse.fs/Contents/Resources/load_osxfuse; else exit 1; fi
           make ALL_TESTS=1 check -C test || (test/filter-suite-log.sh test/test-suite.log; exit 1)
 
+  MemoryTest:
+    runs-on: ubuntu-latest
+
+    #
+    # build matrix for containers
+    #
+    strategy:
+      #
+      # do not stop jobs automatically if any of the jobs fail
+      #
+      fail-fast: false
+
+      #
+      # matrix for type of checking
+      #
+      matrix:
+        checktype:
+          - glibc_debug
+          - sanitize_address
+          - sanitize_memory
+          - sanitize_thread
+          - sanitize_others
+          - valgrind
+
+    container:
+      image: ubuntu:22.04
+
+      options: "--privileged --cap-add SYS_ADMIN --device /dev/fuse"
+
+    env:
+      # [NOTE]
+      # Installation special environment variables for debian and ubuntu.
+      #
+      DEBIAN_FRONTEND:  noninteractive
+
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+
+      - name: Install packages
+        run: |
+          .github/workflows/linux-ci-helper.sh ubuntu:22.04
+
+      - name: Install clang
+        run: |
+          apt-get install -y clang
+          if [ "${{ matrix.checktype }}" = "valgrind" ]; then
+              apt-get install -y valgrind
+          fi
+
+      #
+      # Set CXX/CXXFLAGS and Variables for test
+      #
+      - name: Set variables
+        run: |
+          COMMON_CXXFLAGS='-g -O0 -Wno-cpp -DS3FS_PTHREAD_ERRORCHECK=1'
+          if [ "${{ matrix.checktype }}" = "glibc_debug" ]; then
+              echo "CXXFLAGS=${COMMON_CXXFLAGS} -D_GLIBCXX_DEBUG"  >> $GITHUB_ENV
+          elif [ "${{ matrix.checktype }}" = "sanitize_address" ]; then
+              echo 'CXX=clang++'                                   >> $GITHUB_ENV
+              echo "CXXFLAGS=${COMMON_CXXFLAGS} -fsanitize=address -fsanitize-address-use-after-scope" >> $GITHUB_ENV
+              echo 'ASAN_OPTIONS=detect_leaks=1,detect_stack_use_after_return=1'                       >> $GITHUB_ENV
+          elif [ "${{ matrix.checktype }}" = "sanitize_memory" ]; then
+              echo 'CXX=clang++'                                   >> $GITHUB_ENV
+              echo "CXXFLAGS=${COMMON_CXXFLAGS} -fsanitize=memory" >> $GITHUB_ENV
+          elif [ "${{ matrix.checktype }}" = "sanitize_thread" ]; then
+              echo 'CXX=clang++'                                   >> $GITHUB_ENV
+              echo "CXXFLAGS=${COMMON_CXXFLAGS} -fsanitize=thread" >> $GITHUB_ENV
+              echo 'TSAN_OPTIONS=halt_on_error=1'                  >> $GITHUB_ENV
```

Comment:
This has an actual failure:

```
WARNING: ThreadSanitizer: data race (pid=16744)
  Write of size 4 at 0x7b4c000213e8 by thread T94 (mutexes: write M0, write M1):
    #0 FdEntity::RowFlushMultipart(PseudoFdInfo*, char const*) /__w/s3fs-fuse/s3fs-fuse/src/fdcache_entity.cpp:1627:24 (s3fs+0x173293) (BuildId: 53943963e06f6d42da72c8b1517f40d7dcc5393c)
    #1 FdEntity::RowFlush(int, char const*, AutoLock::Type, bool) /__w/s3fs-fuse/s3fs-fuse/src/fdcache_entity.cpp:1446:18 (s3fs+0x17069d) (BuildId: 53943963e06f6d42da72c8b1517f40d7dcc5393c)
    #2 FdEntity::Flush(int, AutoLock::Type, bool) /__w/s3fs-fuse/s3fs-fuse/src/./fdcache_entity.h:143:82 (s3fs+0xf8ab5) (BuildId: 53943963e06f6d42da72c8b1517f40d7dcc5393c)
    #3 s3fs_flush(char const*, fuse_file_info*) /__w/s3fs-fuse/s3fs-fuse/src/s3fs.cpp:2923:23 (s3fs+0xe6214) (BuildId: 53943963e06f6d42da72c8b1517f40d7dcc5393c)
    #4 <null> <null> (libfuse.so.2+0xf863) (BuildId: 8fcf231a24f8bf85cbcec5ece4ecca31f5337d20)

  Previous read of size 4 at 0x7b4c000213e8 by thread T231:
    #0 FdEntity::IsDirtyNewFile() /__w/s3fs-fuse/s3fs-fuse/src/./fdcache_entity.h:152:64 (s3fs+0xf9620) (BuildId: 53943963e06f6d42da72c8b1517f40d7dcc5393c)
    #1 s3fs_release(char const*, fuse_file_info*) /__w/s3fs-fuse/s3fs-fuse/src/s3fs.cpp:3023:33 (s3fs+0xe68c0) (BuildId: 53943963e06f6d42da72c8b1517f40d7dcc5393c)
    #2 <null> <null> (libfuse.so.2+0xc982) (BuildId: 8fcf231a24f8bf85cbcec5ece4ecca31f5337d20)
```

Could we disable this for now and merge the working parts?

Reply:
Like sanitize_memory, I comment it out and don't run it for now.
Regarding warnings, let's check the code later.